docs/source/audit/implementation.rst

```diff
@@ -0,0 +1,149 @@
+Implementation notes
+====================
+
+.. contents:: Table of Contents
+       :depth: 3
+
+Abstract
+========
+
+This document provides basic information for backend developers regarding the
+auditing implementation in TrueNAS.
+
+
+Specification
+=============
+
+Storage and lifecycle management
+--------------------------------
+The auditing framework implementation is entirely governed by two middlewared
+services:
+
+*auditbackend* - this private service maintains connection to the auditing
+sqlite3 databases and enforces audit record retention periods.
+
+*audit* - this public service provides user-facing APIs that allow querying
+audit records, exporting audit reports, and configuring audit dataset settings and retention period.
+
+Developers must not write code interacting with the audit dataset or auditing
+databases outside of the methods in these two plugins.
+
+1. audit file storage location
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+TrueNAS audit information is written to per-service sqlite3 databases located
+in a dedicated dataset boot device that is mounted at the path /audit. The
```

Comment:
What is `dedicated dataset boot device`? Looks like we have a separate physical device for storing audit data.

Reply:
Apparently I saw a squirrel while writing this.